core/asm/lexer.go

```diff
@@ -254,7 +254,7 @@ func lexInsideString(l *lexer) stateFn {
 
 func lexNumber(l *lexer) stateFn {
 	acceptance := Numbers
-	if l.accept("0") && l.accept("xX") {
+	if l.accept("0") || l.accept("xX") {
```

Comment:
Seems to me the old code was checking for `0x` as a prefix. Why was that behavior bad, and your suggestion (which checks for `0` or `x` as a prefix) good?